src/java.base/share/classes/java/lang/Float16.java

```diff
@@ -859,36 +873,188 @@ public static Float16 sqrt(Float16 radicand) {
+         * of non-finite values by double * and +. Any NaN argument to
+         * fma will lead to a NaN result, infinities will propagate or
+         * get turned into NaN as appropriate, etc.
+         *
+         * One or more zero arguments are also handled correctly,
+         * including the propagation of the sign of zero if all three
+         * arguments are zero.
+         *
+         * The double format has 53 logical bits of precision and its
+         * exponent range goes from -1022 to 1023. The Float16 format
+         * has 11 bits of logical precision and its exponent range
+         * goes from -14 to 15. Therefore, the bit positions
+         * representable in the Float16 format range from the
+         * subnormal 2^(-24), MIN_VALUE, to 2^15, the leading bit
+         * position of MAX_VALUE.
+         *
+         * Consequently, a double can hold the exact sum of any two
+         * Float16 values as the maximum difference in exponents of
+         * Float16 values less than the precision width of double.
+         *
+         * In cases where the numerical value of (a * b) + c is
+         * computed exactly in a double, after a single rounding to
+         * Float16, the result is of necessity correct since the one
+         * double -> Float16 conversion is the only source of
+         * numerical error. The operation as implemented in those
+         * cases would be equivalent to rounding the infinitely precise
+         * value to the result format, etc.
+         *
+         * However, for some inputs, the intermediate product-sum will
+         * *not* be exact and additional analysis is needed to justify
+         * not having any corrective computation to compensate for
+         * intermediate rounding errors.
+         *
+         * The following analysis will rely on the range of bit
+         * positions representable in the intermediate
+         * product-sum.
+         *
+         * For the product a*b of Float16 inputs, the range of
+         * exponents for nonzero finite results goes from 2^(-28)
+         * (from MIN_VALUE squared) to 2^31 (from the exact value of
+         * MAX_VALUE squared). This full range of exponent positions,
+         * (31 -(-28) + 1 ) = 60 exceeds the precision of
+         * double. However, only the product a*b can exceed the
+         * exponent range of Float16. Therefore, there are three main
+         * cases to consider:
+         *
+         * 1) Large exponent product, exponent > Float16.MAX_EXPONENT
+         *
+         * The magnitude of the overflow threshold for Float16 is:
+         *
+         * MAX_VALUE + 1/2 * ulp(MAX_VALUE) =  0x1.ffcp15 + 0x0.002p15 = 0x1.ffep15
+         *
+         * Therefore, any product greater than 0x1.ffep15 + MAX_VALUE
+         * = 0x1.ffdp16 will certainly overflow (under round to
+         * nearest) since adding in c = -MAX_VALUE will still be above
+         * the overflow threshold.
```

Comment:
```suggestion
         * Therefore, for any product greater than or equal to 0x1.ffep15 + MAX_VALUE
         * = 0x1.ffdp16, the "fma" will certainly overflow (under round to
         * nearest), since adding in c = -MAX_VALUE will still be above
         * or at the overflow threshold.
```